server/utils/pagination.ts

```diff
@@ -0,0 +1,82 @@
+export type PaginationPreviousOrNext = {
+  href: string
+  text?: string
+  attributes?: Record<string, string>
+}
+
+export type PaginationItem =
+  | {
+      number: number
+      href: string
+      current?: boolean
+    }
+  | {
+      ellipsis: true
+    }
+
+export type Pagination = {
+  previous?: PaginationPreviousOrNext
+  items?: Array<PaginationItem>
+  next?: PaginationPreviousOrNext
+  landmarkLabel?: string
+}
+
+/**
+ * Produces parameters for GOV.UK Pagination component macro
+ * NB: `page` starts at 1
+ *
+ * Accessibility notes:
+ * - set `landmarkLabel` on the returned object otherwise the navigation box is announced as "results"
+ * - set `previous.attributes.aria-label` and `next.attributes.aria-label` on the returned object if "Previous" and "Next" are not clear enough
+ */
+export function pagination(page: number, pageCount: number, hrefPrefix: string): Pagination {
```

Comment:
really minor, but wonder if `page` could be changed to `currentPage` in this function? found it quite hard to understand what's going on here 😅